packages/core/src/path.mjs

```diff
@@ -802,76 +802,65 @@ Path.prototype.split = function (point) {
   let secondHalf = []
   for (let pi = 0; pi < divided.length; pi++) {
     let path = divided[pi]
+    if (path.ops[0].to.sitsRoughlyOn(point)) {
+      divided[pi].ops[0].to = point.copy()
+      if (pi > 0) {
+        divided[pi - 1].ops[1].to = point.copy()
+      }
+      firstHalf = divided.slice(0, pi)
+      secondHalf = divided.slice(pi)
+      break
+    }
     if (path.ops[1].type === 'line') {
-      if (path.ops[0].to.sitsRoughlyOn(point)) {
-        secondHalf.push(new Path().__withLog(this.log).move(path.ops[0].to).line(path.ops[1].to))
-      } else if (path.ops[1].to.sitsRoughlyOn(point)) {
-        firstHalf.push(new Path().__withLog(this.log).move(path.ops[0].to).line(path.ops[1].to))
-      } else if (pointOnLine(path.ops[0].to, path.ops[1].to, point)) {
+      if (pointOnLine(path.ops[0].to, path.ops[1].to, point)) {
         firstHalf = divided.slice(0, pi)
-        firstHalf.push(new Path().__withLog(this.log).move(path.ops[0].to).line(point))
+        firstHalf.push(new Path().__withLog(this.log).move(path.ops[0].to).line(point.copy()))
         pi++
         secondHalf = divided.slice(pi)
-        secondHalf.unshift(new Path().__withLog(this.log).move(point).line(path.ops[1].to))
+        secondHalf.unshift(new Path().__withLog(this.log).move(point.copy()).line(path.ops[1].to))
+        break
       }
     } else if (path.ops[1].type === 'curve') {
-      if (path.ops[0].to.sitsRoughlyOn(point)) {
-        secondHalf.push(
-          new Path()
-            .__withLog(this.log)
-            .move(path.ops[0].to)
-            .curve(path.ops[1].cp1, path.ops[1].cp2, path.ops[1].to)
+      let t = pointOnCurve(path.ops[0].to, path.ops[1].cp1, path.ops[1].cp2, path.ops[1].to, point)
```

Comment:
We now no longer handle the edge-case where we are splitting on the end of a curve. However, there's a unit test for that and it passes. Plus I don't remember why this was added, so let's just go for it :)